src/sidebar/config/build-settings.js

```diff
@@ -29,129 +31,156 @@ function getAncestorFrame(levels, window_ = window) {
 }
 
 /**
- * Merge client configuration from h service with config from the hash fragment.
+ * Merge configuration sources together to generate `SidebarSettings`
```

Comment:
It's probably easier to look at this whole module instead of reviewing the diff.